src/whylogs/core/statistics/constraints.py

```diff
@@ -438,3 +444,56 @@ def report(self):
         l1 = [(k, v.report()) for k, v in self.value_constraint_map.items()]
         l2 = [(k, s.report()) for k, s in self.summary_constraint_map.items()]
         return l1 + l2
+
+
+class ComplexSummaryConstraint(SummaryConstraints):
```

Comment:
Overall this is nice pattern. I think we need merge defined here if we were to push this (and some checks to make sure we are merging two identical ComplexSummaryConstraints in terms of their components), but let's hold off on the complex constraint approach for the in-between use case, and bring this back if we need it in later constraints.

Reply:
Sure, sounds good.